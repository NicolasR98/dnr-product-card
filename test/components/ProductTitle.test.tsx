import React from 'react';

import renderer from 'react-test-renderer';

import { ProductCard } from '../../src/components/ProductCard';
import { ProductTitle } from '../../src/components/ProductTitle';

import { product1 } from '../data/products';

describe('Tests on ProductTitle', () => { 
    test('Should display correctly', () => {
        const wrapper = renderer.create(
            <ProductTitle title='Custom product'/>
        );

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('Should display correctly the product name', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})
