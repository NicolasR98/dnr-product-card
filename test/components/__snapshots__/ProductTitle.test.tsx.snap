// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on ProductTitle Should display correctly 1`] = `
<span
  className="productDescription undefined"
>
  Custom product
</span>
`;

exports[`Tests on ProductTitle Should display correctly the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee mug - No image
  </span>
</div>
`;
